### PR TITLE
feat(17): Shelf management initial structure

### PR DESCRIPTION
**Pull request by belgithub**

### CODE REVIEW

File: packages/web/src/App.tsx [hunk 1]
```diff
@@ -22,7 +23,8 @@ const App = () => (
       <ConnectedRouter history={history}>
         <Switch>
           <Route path="/signin" component={SignIn} />
-          <Route path="/signout" component={SignOut} />
+          <Route path="/my-shelfs" component={MyFlowerShelfs} />
```
Comment (karolinayanko) [inline packages/web/src/App.tsx]: I'd suggest path simple 'shelfs' and name can also be "My Shelfs"

Comment (karolinayanko) [reply packages/web/src/App.tsx]: and maybe place it in the bottom, not between signin signout

File: packages/web/src/components/AddNewShelfForm/AddNewShelfForm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    const submitButton = getByTestId('submitButton');
+
+    fireEvent.change(nameInput, { target: { value: 'name' } });
+    fireEvent.change(locationInput, { target: { value: 'some location' } });
+    fireEvent.change(descriptionInput, { target: { value: 'cool shelf' } });
+
+    expect(submitButton).toBeEnabled();
+
+    fireEvent.click(submitButton);
+    // expect(fn).toBeCalled();
```
Comment (karolinayanko) [inline packages/web/src/components/AddNewShelfForm/AddNewShelfForm.test.tsx]: Don't forget to uncomment or remove this

File: packages/web/src/components/AddNewShelfForm/AddNewShelfForm.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+export const AddNewShelfForm = ({ onSubmit }: Props) => {
+  const initialValues = {
+    name: '',
+    location: '',
+    description: '',
+  };
+
+  const nameErrorMessage = (name: string): string | null => {
+    if (name.length < 3 || name.length > 15) {
```
Comment (karolinayanko) [inline packages/web/src/components/AddNewShelfForm/AddNewShelfForm.tsx]: "My Awesome Shelf" will not fit max length. I'd suggest change it to common 256 symbols

File: packages/web/src/App.tsx (hunk 1 above)
Comment (dubbha) [reply packages/web/src/App.tsx]: Fixed as agreed with @belgithub

File: packages/web/src/components/AddNewShelfForm/AddNewShelfForm.test.tsx (hunk 2 above)
Comment (dubbha) [reply packages/web/src/components/AddNewShelfForm/AddNewShelfForm.test.tsx]: Fixed as agreed with @belgithub

File: packages/web/src/components/AddNewShelfForm/AddNewShelfForm.tsx (hunk 3 above)
Comment (dubbha) [reply packages/web/src/components/AddNewShelfForm/AddNewShelfForm.tsx]: Fixed as agreed with @belgithub

**Review by dartwlad — APPROVED** [review]

File: packages/web/src/App.tsx [hunk 4]
```diff
@@ -22,7 +22,7 @@ const App = () => (
       <ConnectedRouter history={history}>
         <Switch>
           <Route path="/signin" component={SignIn} />
-          <Route path="/signout" component={SignOut} />
+          <Route path="/signout" component={SignOut} />
```
Comment (karolinayanko) [inline packages/web/src/App.tsx]: redundant space in the end of line

**Review by karolinayanko — APPROVED** [review]

**Review by dubbha — APPROVED** [review]

**Pull request merged**